generators/cmd/predefined/auth_helper.go

```diff
@@ -40,6 +40,17 @@ func authHelper(ac *apiClient, cmd *cobra.Command, args []string) error {
 			AuthKeyID: &providedAuthKeyID,
 			AuthKey:   &providedAuthKey,
 		}
+	} else if providedCommandToProvideCredentials != "" {
+		profile, err := getProfileFromExternalCommand(providedCommandToProvideCredentials)
+		if err != nil {
+			lib.PrintfStderr("unable to get credentials from an external command.\n")
+			return err
+		}
+
+		areq = &authRequest{
+			AuthKeyID: profile.AuthKeyID,
+			AuthKey:   profile.AuthKey,
```

Comment:
The credential information supplied by the command need not be limited to the AuthKeyID and AuthKey pair. That is, the command may supply Email/Password to log in as a root user, or it may supply Operator ID/Username/Password to log in as a SAM user. In order to be able to handle any of these cases, it would be a good idea to copy all the fields from the profile in the authRequest, as we do in the else clause.

コマンドによって供給されるクレデンシャル情報は AuthKeyID と AuthKey のペアに限る必要はないと思います。すなわち、コマンドによってルートユーザーとして認証するための Email / Password が供給されても良いと思いますし、SAM ユーザーとして認証するための Operator ID / Username / Password が供給されてもよいと思います。いずれの場合も対応できるように、authRequest には else 節で行っているように、profile からすべてのフィールドをコピーするようにするのが良いのでは無いでしょうか。

Reply:
@bearmini 
It would certainly be better to be able to receive all the fields used for authentication, as you point out. Thank you!
I'll try to modify so.

Reply:
@bearmini 
I modified the code and pushed it!